app/compile/template/index.js

```diff
@@ -118,7 +135,19 @@ export default async function template(files, options = {}) { // eslint-disable-
 
     let { locales: locales_to_build } = locals
     delete locals.locales
-    locales_to_build = to.array(locales_to_build, /[\s,]+/)
+
+    if (locales_to_build.includes('all')) {
+      locales_to_build = all_locales
+    }
+
+    if (locales_from_folders) {
+      const parts = file.split(path.sep)
+      const file_locale = parts[parts.indexOf('locales') + 1]
+      if (!locales_to_build.includes(file_locale)) {
+        return null
+      }
+      locales_to_build = [ file_locale ]
```

Comment:
This will cause the `project watch --locale eng mex` command to fail because you're reseting the `locales_to_build` to be different. All you should need to do is remove this line.